qcodes/instrument_drivers/Keysight/keysight_34980a.py

```diff
@@ -0,0 +1,217 @@
+                    name = f'slot_{slot}_{module}'
+                    sub_mod = KEYSIGHT_MODELS[module](self, name, slot)
+                    self.module[slot] = sub_mod
+                    self.add_submodule(name, sub_mod)
+                    break
+            if self.module[slot] is None:
+                name = f'slot_{slot}_{module_info}_no_driver'
+                sub_mod_no_driver = KeysightSubModule(self, name, slot)
+                self.module[slot] = sub_mod_no_driver
+                self.add_submodule(name, sub_mod_no_driver)
+                logging.warning(f'can not find driver for {module_info}'
+                                f'in slot {slot}')
+
+    @property
+    def system_slots_info(self):
+        if self._system_slots_info_dict is None:
+            self._system_slots_info_dict = self._system_slots_info()
+        return self._system_slots_info_dict
+
+    def _system_slots_info(self) -> dict:
+        """
+        the command SYST:CTYP? returns the following:
+        Agilent	Technologies,<Model Number>,<Serial Number>,<Firmware Rev>
+        where <Model Number> is '0' if there is no module connected to the
+        given slot
+
+        Returns:
+            a dictionary, with slot numbers as the keys, and model numbers
+            the values
+        """
+        slots_dict = {}
+        keys = ['vendor', 'model', 'serial', 'firmware']
+        for i in range(1, self._total_slot+1):
+            identity = self.ask(f'SYST:CTYP? {i}').strip('"').split(',')
+            if identity[1] != '0':
+                slots_dict[i] = dict(zip(keys, identity))
+        return slots_dict
+
+    def _are_closed(self, channel: str) -> List[bool]:
+        """
+        to check if a channel is closed/connected
+
+        Args:
+            channel (str): example: '(@1203)' for channel between row=2,
+                                    column=3 in slot 1
+                                    '(@sxxx, sxxx, sxxx)' for multiple channels
+
+        Returns:
+            a list of True and/or False
+            True if the channel is closed/connected
+            False if is open/disconnected.
+        """
+        messages = self.ask(f'ROUT:CLOSe? {channel}')
+        return [bool(int(message)) for message in messages.split(',')]
+
+    def _are_open(self, channel: str) -> List[bool]:
+        """
+        to check if a channel is open/disconnected
+
+        Args:
+            channel (str): example: '(@1203)' for channel between row=2,
+                                    column=3 in slot 1
+                                    '(@sxxx, sxxx, sxxx)' for multiple channels
+
+        Returns:
+            a list of True and/or False
+            True if the channel is open/disconnected
+            False if is closed/connected.
+        """
+        messages = self.ask(f'ROUT:OPEN? {channel}')
+        return [bool(int(message)) for message in messages.split(',')]
+
+    def _connect_paths(self, channel_list: str) -> None:
```

Comment:
this is unused, right? if so, please remove it

Reply:
I have removed all the four unused private methods.